synths/core-synths.scd

```diff
@@ -98,6 +98,60 @@ live coding them requires that you have your SuperDirt instance in an environmen
 		}, [\ir, \ir, \ir, \ir, \ir, \ir, \ir, \ir, \kr]).add; // pan can be modulated
 	};
 
+	// this is the time-stretching sample player, based on an overlap-add method
+	(1..SuperDirt.maxSampleNumChannels).do { |sampleNumChannels|
+
+		var name = format("dirt_stretchsample_%_%", sampleNumChannels, numChannels);
+
+		SynthDef(name, { |out, bufnum, sustain = 1, begin = 0, end = 1, speed = 1, endSpeed = 1, freq = 440, pan = 0, timescale = 1|
+
+			var sound, rate, phase, sawrate, numFrames, index, windowIndex, window, timescaleStep;
+			var sound0, sound1, windowSize;
+
+			// playback speed
+			rate = Line.kr(speed, endSpeed, sustain) * (freq / 60.midicps);
+
+			// sample phase
+			// BufSampleRate adjusts the rate if the sound file doesn't have the same rate as the soundcard
+			//phase =  Sweep.ar(1, rate * BufSampleRate.ir(bufnum)) + (BufFrames.ir(bufnum) * begin);
+
+			numFrames = BufFrames.ir(bufnum);
+			// try windows that are 1/29th the width of the result, but not too small or large or it sounds weird
+			windowSize = numFrames * timescale / 29.0;
+			windowSize = windowSize.clip(1000*timescale, BufSampleRate.ir(bufnum)/10);
+			timescaleStep = windowSize / timescale;
+			sawrate = rate * BufSampleRate.ir(bufnum) / (absdif(begin, end) * timescale * numFrames);
```

Comment:
In general: the synth will become a bit more efficient I believe when you refactor so that divisions become multiplications as far as reasonable. This is in particular when the UGens involved are audio rate UGens. Just a suggestion!